script/Makefile

```diff
@@ -267,19 +267,25 @@ test-common: do-build
 test-smoke: do-build
 	FAILED=0; STAGING_RUNTIME_REPO="$(STAGING_RUNTIME_REPO)"; \
 	go test -timeout 10m -v ./e2e/common/main_test.go -tags=integration $(TEST_INTEGRATION_COMMON_LANG_RUN) $(GOTESTFMT) || FAILED=1; \
-	go test -timeout 30m -v ./e2e/common/languages -tags=integration -parallel=$(TEST_COMMON_PARALLEL_COUNT) $(TEST_INTEGRATION_COMMON_LANG_RUN) $(GOTESTFMT) || FAILED=1; \
-	go test -timeout 30m -v \
-		./e2e/common/misc/cron_test.go \
-		./e2e/common/misc/kamelet_test.go \
-		./e2e/common/misc/pipe_test.go \
-	 	-tags=integration -parallel=$(TEST_COMMON_PARALLEL_COUNT) $(TEST_INTEGRATION_COMMON_LANG_RUN) $(GOTESTFMT) || FAILED=1; \
-	go test -timeout 30m -v \
-		./e2e/common/traits/camel_test.go \
-		./e2e/common/traits/container_test.go \
-		./e2e/common/traits/openapi_test.go \
-		./e2e/common/traits/service_test.go \
-	 	-tags=integration -parallel=$(TEST_COMMON_PARALLEL_COUNT) $(TEST_INTEGRATION_COMMON_LANG_RUN) $(GOTESTFMT) || FAILED=1; \
-	exit $${FAILED}
+	if [ $$FAILED -eq 0 ]; then \
```

Comment:
Is this condition skipping the test if the previous failed? I think that, as a smoke test, we want to test them all regardless each test is failing or not as this is mainly run for nightly release purposes.

Reply:
Yes, this allows a contributer to 'quickly' verify that a change set is ready for PR. IMHO, it is sufficient that one test fails for an outcome that is no different when subsequent fail as well. Not skipping ahead would mean to always wait +30min

If you prefer, we could have something similar to this
https://maven.apache.org/surefire/maven-surefire-plugin/examples/skip-after-failure.html